gdal/ogr/ogrsf_frmts/sqlite/ogrsqliteutility.h

```diff
@@ -37,25 +37,29 @@
 #include <set>
 #include <string>
 
-typedef struct
+class SQLResult
 {
-    char** papszResult;
-    int nRowCount;
-    int nColCount;
-    char *pszErrMsg;
-    int rc;
-} SQLResult;
+    public:
+        char** papszResult;
+        int nRowCount;
+        int nColCount;
+        char *pszErrMsg;
+        int rc;
+
+        SQLResult ();
+        virtual ~SQLResult ();
```

Comment:
I don't think the destructor needs to be virtual as there's little reason that this class should be derived. Actually making it final would be clearer.
And having deleted copy constructor and assignment operators would also be good as we have pointer members in it.